lib/zendesk_apps_support/validations/secrets.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+
+module ZendeskAppsSupport
+  module Validations
+    module Secrets
+      SECRET_KEYWORDS = %w[
+        pass password secret secretToken secret_token auth_key
+        authKey auth_pass authPass auth_user AuthUser username api_key
+      ].freeze
+
+      class << self
+        def call(package)
+          compromised_files = []
+
+          # Rely on the regexes used by truffleHog (https://github.com/dxa4481/truffleHog) to detect application secrets
+          trufflehog_regexes = JSON.parse(Net::HTTP.get(URI.parse('https://raw.githubusercontent.com/dxa4481/truffleHogRegexes/master/truffleHogRegexes/regexes.json'))).freeze
+
+          package.text_files.each do |file|
+            contents = file.read
+
+            trufflehog_regexes.each do |secret_type, regex_str|
```

Comment:
Can we `Regexp.union` these regexes? We'd lose the ability to report on what type of secret was found, but it'd probably be obvious to the developer, right?

Reply:
Doing that seems like a very minor increase in code length + speed to validate vs a potentially very useful pointer to the secret for the app developer. I'd prefer to keep the type information in here.

Reply:
Sure, I agree that this does provide more information to the app developer. Happy to keep it in here.